cfme/middleware/__init__.py

```diff
@@ -1,8 +1,54 @@
 from cfme.web_ui import CheckboxTable, toolbar as tb
 from functools import partial
 
+from cfme.web_ui import Table
+
 cfg_btn = partial(tb.select, 'Configuration')
 mon_btn = partial(tb.select, 'Monitoring')
 pol_btn = partial(tb.select, 'Policy')
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+
+
+class Validatable(object):
+    """
+    Class which Middleware provider and other middleware pages must extend
+    to be able to validate properties values shown in summary page.
+
+    """
+
+    """
+    Tuples which first value is the provider class's attribute name,
+    the second value is provider's UI summary page field key.
+
+    Should have values in child classes.
+
+    """
+    property_tuples = []
+
+    def properties_ui(self):
+        property_ui = {}
+        for row in Table('//*[@id="main-content"]//div[1]/table').rows():
```

Comment:
* Root locators should never live inside the code
* I see this is a copied locator from browser, it is preferable that you do it manually. Eg. if there is no other table in the div, then you can make the locator look like this:

```
//div[@id="main-content"]//table
```